### ISSUE TITLE
question

### ISSUE THREAD

**Issue by oscar-lima** [phuicy/downward#1]
I am trying to make your code work but i am unable, maybe you can help me. this is what i did:

cloned your repo, went to the readme install dependencies, then launched:

```
    roslaunch planner planner.launch
```

then i prepared a client:

```
#! /usr/bin/env python
import rospy
import roslib
import actionlib
import sys

from downward_msgs.msg import downAction
from downward_msgs.msg import downActionGoal

timeout = 10

pddl_domain = "/home/oscar/Documents/pddl_problems/cleaning_robot/domain.pddl"
pddl_problem = "/home/oscar/Documents/pddl_problems/cleaning_robot/problems/p01.pddl"

if __name__ == '__main__':
    rospy.init_node('planner_server_client_tester')
    rospy.loginfo("planner server client tester node initialized...")
    client = actionlib.SimpleActionClient('planner_server', downAction)
    rospy.loginfo("wating for server -planner_server- to respond...")
    client.wait_for_server()
    rospy.loginfo("server is there... making goal message")
    goal = downActionGoal()
    goal.goal.domain = pddl_domain
    goal.goal.problem = pddl_problem
    rospy.loginfo("sending goal to server...")
    client.send_goal(goal.goal)
    rospy.loginfo("wating for server to resolve goal...")
    client.wait_for_result(rospy.Duration.from_sec(int(timeout)))
    rospy.loginfo("print outcome...")
    print client.get_result()
```

---

my test domain look like this:

```
(define (domain cleaning_robot)

  (:requirements
    :typing
  )

  (:types
    location
    robot
  )

  (:predicates
    (at ?r - robot ?l - location)   ; robot r? is at location l?
    (clean ?l - location)           ; location ?l is clean
  ) 

  (:action move
    :parameters (?r - robot ?source ?destination - location)
    :precondition (at ?r ?source)
    :effect (and    ( not (at ?r ?source))
                    (at ?r ?destination)
            )
  )

  (:action clean
    :parameters (?r - robot ?l - location)
    :precondition (and (at ?r ?l) (not(clean ?l)))
    :effect (clean ?l)
  )
)
```

---

As you can see is a simple test domain of a cleanning robot

problem definition looks like this:

```
(define (problem p01)

  (:domain cleaning_robot)

  (:objects 
    ghost - robot
    locA locB - location

  )

  (:init 
    (at ghost locA)
    (not(clean locA))
    (not(clean locB))

  ) 

  (:goal
    (  and (at ghost locA)
      (clean locB)
      (clean locA)
    )
  )
 )
```

---

if I test the domain with the ffha planner (from ros : http://wiki.ros.org/ffha) 

```
    rosrun ffha ffha -o domain.pddl -f problems/p01.pddl
```

it gives solution, which means that domain.pddl and problem definition are ok.

solution from ffha to the proposed problem:

```
ff: found legal plan (steps: 4) as follows
   0: (CLEAN GHOST LOCA)
   1: (MOVE GHOST LOCA LOCB)
   2: (CLEAN GHOST LOCB)
   3: (MOVE GHOST LOCB LOCA)
```

---

this is the error that i get when i run the client :

```
oscar@oscar-MacBook:~ $ roslaunch planner planner.launch 
... logging to /home/oscar/.ros/log/83f69c40-d249-11e5-9fe9-5c969d8d2b33/roslaunch-oscar-MacBook-5707.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://192.168.178.26 :44126/

SUMMARY
========

PARAMETERS
 * /rosdistro
 * /rosversion

NODES
  /
    planner_server (planner/planner_server)
    preprocessor (preprocess/preprocess_server)
    searcher (search/search_server)
    translator (translate/translate.py)

auto-starting new master
process[master]: started with pid [5721]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 83f69c40-d249-11e5-9fe9-5c969d8d2b33
process[rosout-1]: started with pid [5734]
started core service [/rosout]
process[translator-2]: started with pid [5746]
[INFO] [WallTime: 1455364891.416311] Translator ready
process[preprocessor-3]: started with pid [5757]
[ INFO] [/preprocessor]: Ready to preprocess
process[searcher-4]: started with pid [5769]
[ INFO] [/searcher]: Ready to Search
process[planner_server-5]: started with pid [5780]
[ INFO] [/planner_server]: Received goal
[ INFO] [/planner_server]: translating to SAS
[ERROR] [WallTime: 1455364905.333172] Error processing request: Expected '(', got /home/oscar/documents/thesis/pddl/pddl_problems/cleaning_robot/domain.pddl.
['Traceback (most recent call last):\n', '  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 623, in _handle_request\n    response = convert_return_to_response(self.handler(request), self.response_class)\n', '  File "/home/oscar/ros_ws/robocup/src/downward/translate/scripts/translate.py", line 589, in translator\n    task = pddl.open(req.problem,req.domain)\n', '  File "/home/oscar/ros_ws/robocup/src/downward/translate/scripts/pddl/pddl_file.py", line 20, in open\n    domain_pddl = parser.parse_nested_list(domain_filename)\n', '  File "/home/oscar/ros_ws/robocup/src/downward/translate/scripts/pddl/parser.py", line 12, in parse_nested_list\n    raise ParseError("Expected \'(\', got %s." % next_token)\n', "ParseError: Expected '(', got /home/oscar/documents/thesis/pddl/pddl_problems/cleaning_robot/domain.pddl.\n"]
[ERROR] [/planner_server]: Service call failed: service [/translator] responded with an error: error processing request: Expected '(', got /home/oscar/documents/thesis/pddl/pddl_problems/cleaning_robot/domain.pddl.
[ERROR] [/planner_server]: Unable to translate
```

---

so basically the error says that:

Expected '(', got /home/oscar/documents/thesis/pddl/pddl_problems/cleaning_robot/domain.pddl

expects (

mm interesting, but i checked your code and it looks for filename in the passed string not pddl

Any idea what i am making wrong here?

thanks!!

**Comment by phuicy** [comment]
It seems to run at my end. Have you run the examples, in test?

**Comment by oscar-lima** [comment]
ok thanks for the help!, basically for some pddl domains it works and for some others not. out of 5 different domains that i have works for 2. :D

**Comment by phuicy** [comment]
Fast downward, takes a specific [PDDL](http://www.fast-downward.org/PddlSupport) version.
